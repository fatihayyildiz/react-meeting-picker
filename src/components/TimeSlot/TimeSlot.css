.time-slot-box {
	padding: 5px;
	text-align: center;
	border: 1px solid #e1e1e1;
	border-radius: 2px;
	margin: 5px auto 0 !important;
	cursor: pointer;
	background-color: white;
}

.disabled {
	background-color: rgba(209, 209, 209, 0.65);
	cursor: not-allowed;
}

.selected {
	background-color: rgba(111, 99, 229, 0.75);
}
