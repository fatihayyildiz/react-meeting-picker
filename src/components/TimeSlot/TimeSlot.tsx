import React from 'react';
import 'components/TimeSlot/TimeSlot.css';
import classNames from 'classnames';
import { useDispatch } from 'react-redux';
import { AppDispatch } from 'store/store';
import slotSlice from 'store/slots/slice';

export type TimeSlotProps = {
	id: number;
	day: string;
	start_time: string;
	end_time: string;
	available?: boolean;
	selected?: boolean;
};

const TimeSlot = ({
	id,
	day,
	start_time,
	end_time,
	available = true,
	selected = false,
}: TimeSlotProps) => {
	const dispatch = useDispatch<AppDispatch>();

	const onSlotClick = () => {
		dispatch(
			slotSlice.actions.selectSlotForCompany({ id, day, start_time, end_time })
		);
	};

	return (
		<div
			className={classNames('row', 'time-slot-box', {
				available: available,
				selected: selected,
			})}
			onClick={() => onSlotClick()}
		>
			<div className="col-4">{start_time}</div>
			<div className="col-2">-</div>
			<div className="col-4">{end_time}</div>
		</div>
	);
};

export default TimeSlot;
