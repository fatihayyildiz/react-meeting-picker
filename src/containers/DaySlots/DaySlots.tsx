import React from 'react';
import { TimeSlotProps } from 'components/TimeSlot/TimeSlot';
import TimeSlot from 'components/TimeSlot/TimeSlot';
import 'containers/DaySlots/DaySlots.css';

export type DaySlotsProps = {
	day: string
	slots: Array<TimeSlotProps>
}

const DaySlots = ({ day, slots }: DaySlotsProps) => {
	return (
		<div className='container centered-box'>
			<div className="row">
				<div className="col-12">{day}</div>
			</div>
			<div className="row g-0 centered-box">
				{
					slots.map((slot,index)=>{
						return <TimeSlot key={`slot-${index}`} {...{
							start_time: slot.start_time,
							end_time: slot.end_time
						}} />
					})
				}
			</div>
		</div>
	);
};

export default DaySlots;
