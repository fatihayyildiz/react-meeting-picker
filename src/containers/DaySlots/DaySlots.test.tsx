import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DaySlots from 'containers/DaySlots/DaySlots';
import SingleFormattedCompany from 'mocks/resources/single_formatted_company.json';
import { TimeSlotProps } from 'components/TimeSlot/TimeSlot';

afterEach(cleanup);

const testProps = {
	id: 1,
	day: SingleFormattedCompany[0].days[0].day,
	slots: SingleFormattedCompany[0].days[0]
		.slots as unknown as Array<TimeSlotProps>,
};

test('Day slots should match with snapshot', () => {
	// First render component and get as fragment
	const { asFragment } = render(
		<DaySlots
			{...{ day: testProps.day, slots: testProps.slots, id: testProps.id }}
		/>
	);

	// Compare with saved snapshot
	expect(asFragment()).toMatchSnapshot();
});

test('Day slots renders classnames correctly', () => {
	// First render component with props and get as fragment
	const { container } = render(
		<DaySlots
			{...{ day: testProps.day, slots: testProps.slots, id: testProps.id }}
		/>
	);

	// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
	const centeredBox = container.getElementsByClassName('centered-box');

	// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
	const row = container.getElementsByClassName('row');

	// Check classes
	expect(centeredBox.length).toBe(2);

	// Check rows also contains time slots row classnames
	expect(row.length).toBe(4);
});
