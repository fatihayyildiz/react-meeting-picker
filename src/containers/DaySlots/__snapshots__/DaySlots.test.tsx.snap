// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day slots should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="container centered-box"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        18.03 - Monday
      </div>
    </div>
    <div
      class="row g-0 centered-box no-border"
    >
      <div
        class="row time-slot-box"
      >
        <div
          class="col-4"
        >
          08:00
        </div>
        <div
          class="col-2"
        >
          -
        </div>
        <div
          class="col-4"
        >
          09:00
        </div>
      </div>
      <div
        class="row time-slot-box"
      >
        <div
          class="col-4"
        >
          10:00
        </div>
        <div
          class="col-2"
        >
          -
        </div>
        <div
          class="col-4"
        >
          11:00
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
